docs/troubleshooting.md

```diff
@@ -0,0 +1,73 @@
+# Troubleshooting
+
+If you have problems, try to rebuild when compiling with `-DCMAKE_BUILD_TYPE=Debug`.
+In particular also rebuild the index.
+The release build assumes machine written words- and docsfiles and omits sanity checks for the sake of speed.
+
+## Run End-to-End Tests
+
+QLever includes a simple mechanism for testing the entire system from
+from building an index to running queries in a single command.
+
+In fact this End-to-End Test is run on Travis CI for every commit and Pull
+Request. It is also useful for local development however since it allows you to
+quickly test if something is horribly broken.
+
+Just like QLever itself the End-to-End Tests can be executed from a previously
+build QLever container
+
+**Note**: If you encounter permission issues e.g. if your UID is not 1000 or you
+are using docker with user namespaces, add the flag `-u root` to your `docker
+run` command or do a `chmod -R o+rw e2e_data`
+
+    docker build -t qlever .
+    docker run -it --rm -v "$(pwd)/e2e_data:/app/e2e_data/" --name qlever-e2e --entrypoint e2e/e2e.sh qlever
+
+## Converting old Indices For Current QLever Versions
```

Comment:
`old` should also be capitalized, as `Current` is capitalized.

Reply:
I've made "current" lower case. Funnily enough the German "upper case nouns and the start of a sentence" algorithm is a valid title casing in English and actually the one I was aiming for